xds/src/main/java/io/grpc/xds/ClusterResolverLoadBalancer.java

```diff
@@ -211,30 +209,35 @@ private void handleEndpointResourceUpdate() {
       List<EquivalentAddressGroup> addresses = new ArrayList<>();
       Map<String, PriorityChildConfig> priorityChildConfigs = new HashMap<>();
       List<String> priorities = new ArrayList<>();  // totally ordered priority list
-      boolean allResolved = true;
+      Status endpointNotFound = Status.OK;
       for (String cluster : clusters) {
         ClusterState state = clusterStates.get(cluster);
-        if (!state.resolved) {
-          allResolved = false;
-          continue;
+        // Propagate endpoints to the child LB policy only after all clusters have been resolved.
+        if (!state.resolved && state.status.isOk()) {
+          return;
```

Comment:
For dns cluster, is there a timeout waiting for the resolution result?

Reply:
Nope, [JNDI](https://docs.oracle.com/javase/7/docs/technotes/guides/jndi/jndi-dns.html) has its built-in timeout/backoff mechanism and it's configured at JVM level (by default initial timeout of 1 sec and max of 4 retries). So we'd assume DNS resolution will finish in a certain amount of time, doing an extra timeout measurement is premature.

Reply:
If EDS endpoints are immediately available but DNS is not responding, why do we wait until DNS timeout (that will be at least 4 sec by default, could be longer if there are more DNS servers or the timeout is configured longer)? That's bad for performance. Instead of waiting for resolution result of all clusters, is it possible to just update endpoints as soon as getting resolution result for clusterA, and not to update endpoints of clusterB until the resolution for clusterA has a result or error?

Reply:
Obsolete.